lib/logger.js

```diff
@@ -48,7 +48,7 @@ function logger(options) {
             winston.format.timestamp(),
             winston.format.colorize(),
             winston.format.printf((info) => {
-                return `${info.timestamp} - ${info.level}: ${info.message}` + (info.meta ? `\n${util.inspect(info.meta, false, 4)}` : '');
+                return `${info.timestamp} - ${info.level}: ${info.message}${info.meta ? `\n${util.inspect(info.meta, false, 4)}` : ''}${info.stack ? `\n${info.stack}` : ''}`;
```

Comment:
This may be better as a custom formatter but I'm not sure. I got this working before using adding a [custom formatter](https://stackoverflow.com/questions/47231677/how-to-log-full-stack-trace-with-winston-3) and adding it to the logger like this:
```
    const errorStackFormat = winston.format(info => {
        if (info instanceof Error) {
            return Object.assign({}, info, {
                stack: info.stack,
                message: info.message
            })
        }
        return info
    });

   // etc.

   options.console.format = winston.format.combine(
       errorStackFormat(),
       winston.format.splat(),
       winston.format.timestamp(),
       winston.format.colorize(),
       winston.format.printf((info) => {
           return `${info.timestamp} - ${info.level}: ${info.message}` + (info.meta ? `\n${util.inspect(info.meta, false, 4)}` : '');  // eslint-disable-line prefer-template
       })
   );
```

Reply:
@nickt-turner Where would be the best place to put a feature flag in this, you reckon?

@nigelelliott This looks like a nice way to do it, what do you think?

Reply:
there is no need for a custom formatter since you can just check for the existence of `info.stack` which will be undefined for anything but an error object

Reply:
See my comment from today, above.  I would argue that it is inappropriate for the logging library to be making content decisions about the content to be logged.

Maybe I've already provided plenty of context in my logs and don't need or want the error stack, but I still want to log the occurence of the Error.  Why should the logging tool then push this extra content on me?  I didn't ask for it, so it shouldn't happen.  It should only strictly log what you've asked for in the message.